moonstone/analysis/diversity/alpha.py

```diff
@@ -159,6 +162,18 @@ def visualize_groups(
             groups=groups,
         )
 
+    def compare_groups(
+        self, metadata_df: pd.DataFrame, group_col: str, stat_test: str = 'mann_whitney_u',
+        plotting_options: dict = None,
+        show_visualization: Optional[bool] = False, output_visualization_file: Optional[str] = False
+    ):
+        method = f"{stat_test}_group"
+        self.stat_test_group_matrix = eval(method+'(self.alpha_diversity_indexes, metadata_df[group_col])')
```

Comment:
It is better to avoid using `eval` when it is not necessary and that an alternative exists.
You can have a look at: https://www.journaldev.com/22504/python-eval-function#security-risks-with-eval-function for instance.

Otherwise, more globally, I think you could have used a more simple approach:

* Having the Mann Whitney in statistical test handling two series and returning the pvalue (or directly use the one from `scipy` when needed)
* Handle in this method the generation of the different groups and pvalue matrix

In the end if the code works this way and perform what you need, we can leave it like this and modify it later if necessary, this was just for you to know